src/audio/volume/volume_generic.c

```diff
@@ -61,17 +61,27 @@ static void vol_s24_to_s24(struct comp_dev *dev, struct audio_stream *sink,
 	uint32_t channel;
 	uint32_t buff_frag = 0;
 
-	/* Samples are Q1.23 --> Q1.23 and volume is Q8.16 */
+	/* Samples are Q1.23 --> Q1.23 and volume is Q8.16 (IPC3)/Q1.23 (IPC4)
+	 */
 	for (i = 0; i < frames; i++) {
 		for (channel = 0; channel < sink->channels; channel++) {
 			src = audio_stream_read_frag_s32(source, buff_frag);
 			dest = audio_stream_write_frag_s32(sink, buff_frag);
 
 			*dest = vol_mult_s24_to_s24(*src, cd->volume[channel]);
 
+#if CONFIG_IPC_MAJOR_4
```

Comment:
Yep, this could be utilized with IPC3 and other too.

Reply:
not sure if I understand correctly, do you want to have CONFIG_PEAK_VOL along with CONFIG_VOLUME on the audio components list?
CONFIG_VOLUME default=y for IPC3
CONFIG_PEAK_VOL default=y for IPC4

and replace all IPC4 #ifdefs with CONFIG_PEAK_VOL?

Reply:
@lgirdwood please clarify :)

Reply:
Sorry, for delay. I mean we can have the feature as option for either IPC3/4 where we report back the peak volume data (it can be left empty for IPC3 though today as more work has to be done here). 
So CONFIG_VOLUME is used by IPC3 and IPC4 (user can change teh volume) and CONFIG_PEAK_VOL (users get information about the peak volume).

Reply:
Got it, thanks. Added CONFIG_PEAK_VOL, currently depends on IPC4, limitation can be removed later when IPC3 support added